### PR TITLE
Add two new checker classes

### PR DESCRIPTION
**Pull request by MuriloDalRi**
Add two new checker classes to which make sure the available locales
defined in each app correspond to the locale files that are present.

[Trello](https://trello.com/c/DQXj8Ce6/2779-add-check-to-compare-available-locales-to-actual-locale-files-3)

### CODE REVIEW

File: lib/rails_translation_manager/locale_checker/actual_available_locales.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class ActualAvailableLocales < BaseChecker
+    include LocaleCheckerHelper
+  
+    def report
+        diff = compare
+        "- \e[31m[ERROR]\e[0m Please check the following locale files exist: '#{diff}' <link to future documentation>" unless diff.empty?
```
Comment (ChrisBAshton) [inline lib/rails_translation_manager/locale_checker/actual_available_locales.rb:8]: We already know the locale files don't exist - I'd reword the error message to something more useful. I'd also split out each missing locale, like we've done for the [other checkers](https://github.com/alphagov/rails_translation_manager/blob/52d262b2b6e246e3570afbb18218d5840736d3bf/lib/rails_translation_manager/locale_checker/incompatible_plurals.rb#L12-L32).

I'd suggest the output should look something like:

```
No locale files detected for:

- pt

Either create these locale files or remove these locales from your I18n `available_locales` config.
```

File: lib/rails_translation_manager/locale_checker/actual_available_locales.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def available_locales
+        I18n.available_locales
+    end
+  
+    def actual_locales
+        all_locales.map { |l| l[:locale] }
+    end
+
+    def compare
+        available_locales - actual_locales | actual_locales - available_locales
```
Comment (ChrisBAshton) [inline lib/rails_translation_manager/locale_checker/actual_available_locales.rb:22]: Looks like you've considered two scenarios:

1. Where a locale is declared in `available_locales`, but has no locale file
2. Where a locale has a locale file, but is not declared in `available_locales`

The use of the bitwise OR operator is quite neat here too (although I did have to try it out in IRB to figure out what it's doing!).

The PR is currently missing a test for 2), so we should add that. I also think each of the two scenarios should have its own error message, as they are checking different things. Really I think these should be implemented as two different checkers, in two commits. (There will be a bit of repetition between them, but that's fine - we made the same compromise on `missing_english_locales` vs `missing_foreign_locales`).

**Review by ChrisBAshton — COMMENTED** [review]
Good start, Murilo - but I think we'll need to split it into two checkers. See below:

File: lib/rails_translation_manager/locale_checker/actual_available_locales.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class ActualAvailableLocales < BaseChecker
+    include LocaleCheckerHelper
```
[the remark this answers is not in the document]
Comment (MuriloDalRi) [reply lib/rails_translation_manager/locale_checker/actual_available_locales.rb:4]: Oops! Have we not set up Rubocop on this repo?

File: lib/rails_translation_manager/locale_checker/actual_available_locales.rb (hunk 1 above)
Comment (MuriloDalRi) [reply lib/rails_translation_manager/locale_checker/actual_available_locales.rb:8]: Yeah I agree the message could be improved, I'll change to something like your suggestion 👍

File: lib/rails_translation_manager/locale_checker/actual_available_locales.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply lib/rails_translation_manager/locale_checker/actual_available_locales.rb:4]: We have not! A nice stretch-goal of this work would be to enable Rubocop, if you fancy it 💪

File: spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        {
+          locale: :pt,
+          keys: %w[browse.same_key]
+        }
+      ]
+    end
+
+    let(:actual_locales) do
+      [:en, :cy]
+    end
```
Comment (ChrisBAshton) [inline spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb]: Where is this being used? 🤔

File: spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          locale: :en,
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :cy,
+          keys: %w[browse.same_key]
+        }
+      ]
+    end
+
```
Comment (ChrisBAshton) [inline spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb:55]: As with the other test, I think it would be a good idea to make it explicit that `en`/`cy` exist, and not rely on the values set in the spec_helper.rb. (I think in a future refactor we'll want to get rid of that, tbh!)

**Review by ChrisBAshton — COMMENTED** [review]
Thanks Murilo, it's coming along (and as pointed out by Jonathan, the Rubocop work is out of scope for this PR 👍 ). A few comments about the tests below:

File: lib/rails_translation_manager/locale_checker/missing_declared_locales.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    \e[31m[ERROR]\e[0m No locale files detected for:
+
+    #{locales.join("\n\n")}
+
+    \e[1mEither create these locale files or remove these locales from your I18n `available_locales` config.\e[22m
+    OUTPUT
+  end
+
+  def available_locales
+    I18n.available_locales
```
[the remark this answers is not in the document]
Comment (MuriloDalRi) [reply lib/rails_translation_manager/locale_checker/missing_declared_locales.rb]: Yes, that's a leftover from the original implementation with a single checker. I've fixed it now

File: spec/rails_translation_manager/locale_checker/undeclared_locale_files_spec.rb [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+          cy
+
+          \e[1mEither declare these locale files or remove them from your locale files directory.\e[22m
+          OUTPUT
+        )
+    end
+  end
+
+  context "when there aren't undeclared locales" do
+    let(:all_locales) do
```
[the remark this answers is not in the document]
Comment (MuriloDalRi) [reply spec/rails_translation_manager/locale_checker/undeclared_locale_files_spec.rb]: Done!

File: spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :cy,
+          keys: %w[browse.same_key]
+        },
+        {
+          locale: :pt,
+          keys: %w[browse.same_key]
+        }
```
[the remark this answers is not in the document]
Comment (MuriloDalRi) [reply spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb]: Fixed

File: spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb (hunk 4 above)
Comment (MuriloDalRi) [reply spec/rails_translation_manager/locale_checker/missing_declared_locales_spec.rb]: I've removed it

**Comment by MuriloDalRi** [comment]
I think I've addressed all your comments, @ChrisBAshton

**Comment by ChrisBAshton** [comment]
@MuriloDalRi looks like the fixes have been applied to the second commit, but they've not been fixed up to the first commit. Could you take another look please?

**Comment by MuriloDalRi** [comment]
> @MuriloDalRi looks like the fixes have been applied to the second commit, but they've not been fixed up to the first commit. Could you take another look please?

Does it look ok now?

**Review by ChrisBAshton — APPROVED** [review]
Perfect ⭐ thanks for addressing all the comments!
